include/aws/common/hash_table.h

```diff
@@ -103,7 +103,7 @@ typedef uint64_t(aws_hash_fn)(const void *key);
  * keys, but note that the same type is used for a function that compares
  * two hash table values in aws_hash_table_eq.
  *
- * Equality functions used in a hash table must be reflexive (i.e., a == b if
+ * Equality functions used in a hash table must be symmetric (i.e., a == b if
```

Comment:
should we call out that this function must implement equivalence relationship? i.e. be reflexive (a == a), symmetric (a == b => b == a), transitive (a == b, b == c => a == c) and consistent (result does not change with time)

Reply:
added this change